### DISCUSSION TITLE
Bounce address identified as: HK_RANDOM_ENVFROM from SpamAssasin

### DISCUSSION THREAD

**Discussion by maxok** [postalserver/postal#1432] [Help with using Postal]
Hello,

I have been doing some testing of emails sent from Postsal through mail-tester.com and have received -0.626 score, because of bounce address being identified as - HK_RANDOM_ENVFROM - (Envelope sender username looks random).

My bounce address: sxdlzr@psrp.example.com (actual host being changed by example.com)

Is it possible to change the bounce address to something more meaningful, so it is not identified as random? 

Below is the bounce address from SendGrid, which is not being identified as random by SpamAssasin:

bounces+12215969-e62a-{email_username}={email_domain}@{host}

{host} - postal server host. Ex: postal.example.com
{email_username} - user, which is sending the email. Ex: mike (In case of mike@example.com)
email_domain - email domain. Ex: example.com

Screenshot also enclosed.

![Screen Shot 2019-12-14 at 9 58 24 AM](https://user-images.githubusercontent.com/1223466/70844311-ac022b00-1e58-11ea-8972-2335c7fe56c3.png)

**Comment by anunemployedceo** [comment]
Make sure in Settings->Advance Settings->Allow Sender Header is set to Yes

**Comment by maxok** [comment]
I did what you have suggested @ehtesham000, restarted postal and tested again, but the issue is the same. Mail-tester still shows "Envelope sender username looks random".

**Comment by willpower232** [comment]
I can't replicate your problem, how are you sending messages through Postal?

**Comment by maxok** [comment]
@willpower232, I have setup Postal on my server with 2 IPs, one for the Web and second for fast-track. So I am sending email through Apple Mail client with Postal SMTP.

I think, the issue is with the way Postal is generating bounce address for its tracking. In my case, it is being down as:

token@psrp.example.com

token - is generating by Postal for each account.

With this configuration, SpamAssasin is considering, that the bounce address is randomly generated and giving negative score for that.

**Comment by willpower232** [comment]
Postal does that for all outgoing messages and my testing does not get the same error.

![image](https://user-images.githubusercontent.com/1619102/70864376-23af8300-1f49-11ea-8480-c47108f846bb.png)

Postal was not intended to send emails from a desktop email client so this isn't an ideal use case.

Are you able to share the bit of your email address before the `@`?

**Comment by maxok** [comment]
Very strange. I have just did another test without putting link or tracking opens and still receive HK_RANDOM_ENVFROM. Enclosed please find raw message.

My address before @ is malkhaz, but I have also tried doing it with another account david and still same issue.

[mail-tester-raw.txt](https://github.com/postalhq/postal/files/3966611/mail-tester-raw.txt)

**Comment by willpower232** [comment]
Very odd indeed. Are you using any relays at all? Do you get the same problem if you send a message from the Postal interface?

**Comment by maxok** [comment]
I just tried the same from Postal interface with same subject and same message and got exactly the same HK_RANDOM_ENVFROM Envelope sender username looks random.

**Comment by rohitkatiyar1984** [comment]
Did we have any resolution for this??  I get the same error!

**Comment by maxok** [comment]
Not really. I tried a few things, but nothing did help, so I left it as it is

**Comment by tiltec** [comment]
The local part of the bounce address is the mail server token (code here: https://github.com/postalhq/postal/blob/ed7889a40e1d777f1d80821043562ab18b04766e/lib/postal/smtp_sender.rb#L119). As far as I know, it is generated when you create a new mail server inside an organization. It is probably random.

In this way, SpamAssassin is correct, but it didn't complain in my case. Maybe you can work around this by creating more mail servers until one isn't recognized as random?
I guess it might also be useful to understand how SpamAssassin randomness detection works.

**Comment by olegbliaher** [comment]
Same issue exactly here with one of my organizations/domains. How stupid would it be to simple the relevant `token` in the database/server table to something like "bounces"? Just so that spamassassin stops complaining...

**Comment by willpower232** [comment]
I think the server token has several uses throughout the system so whilst you could edit the database and set a more human-readable token, you should make sure that the token is still unique within your Postal database.

**Comment by sheteajay** [comment]
I am having the same issue. Is it because I am using test@domainname.com to send an email? or is it because of my bounce address?

**Comment by sheteajay** [comment]
Any advice to fix the score?
![Screenshot 2021-06-13 221408](https://user-images.githubusercontent.com/85721355/121815556-b511c980-cc94-11eb-96c0-636a2bc0f997.png)

**Comment by willpower232** [comment]
What is the from address you are using? i.e. what is the whatever in whatever@example.com for you?

**Comment by GandalfGrey77** [comment]
I have the same isue with this. It was becouse I delate previous mail server in postal and make it again with the same domain. After that The issue appear. 
To fix this I just made another mail server in Postal with different name server (just add "2" to the name). 
And I connect the same domain. It works. 

thx willpower232 ! you were close to solution

**Comment by solracsf** [comment]
To help solving this (by developpers), please check https://stackoverflow.com/questions/45014530/understanding-spamassassin-hk-random-regex

Maybe It could help write better "randoms" not detected as is by SpamAssassin?

**Reply by willpower232** [reply]
thats certainly interesting information and matches the original problem, worth taking a look at I'd say

**Reply by cameronbraid** [reply]
One workaround could be to have a dictionary of real words and then generate a token using X words from the dictionary .. though you would have to check for length violations for the maibox name.  Alternatively just use a random single word and check it for uniqueness against all other tokens already generated

**Reply by mjnaderi** [reply]
Is there any progress on this? @willpower232

**Comment by sheteajay** [comment]
Why am I receiving these emails?
